# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/build/nsLayoutModule.cpp
# Commit: 2663311a1b62
# Full Hash: 2663311a1b628c835b7c557d453797080c92d25d
# Author: Yaron Tausky <ytausky@mozilla.com>
# Date: 2019-12-09 21:50:19
# Regressor Bug: 1593246
# File Overlap Count: 1
# Description:
#   Bug 1593246 - Part 2: Give SessionStorageManager a reference to BrowsingContext r=sg
#   
#   This reference is necessary when sending session storage data for
#   all browsing context to the parent process. Note that it entails
#   making SessionStorageManager a cycle collection participant, since
# ==============================================================================

diff -r d9f0d827e28d -r 2663311a1b62 layout/build/nsLayoutModule.cpp
--- a/layout/build/nsLayoutModule.cpp	Mon Dec 09 13:04:51 2019 +0000
+++ b/layout/build/nsLayoutModule.cpp	Thu Dec 05 17:51:59 2019 +0000
@@ -241,7 +241,7 @@
 
 nsresult SessionStorageManagerConstructor(nsISupports* aOuter, REFNSIID aIID,
                                           void** aResult) {
-  RefPtr<SessionStorageManager> manager = new SessionStorageManager();
+  RefPtr<SessionStorageManager> manager = new SessionStorageManager(nullptr);
   return manager->QueryInterface(aIID, aResult);
 }
 
